plugins/outputs/http/http.go

```diff
@@ -0,0 +1,273 @@
+package http
+
+import (
+	"bytes"
+	"context"
+	ejson "encoding/json"
+	"errors"
+	"fmt"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/serializers"
+	"github.com/influxdata/telegraf/plugins/serializers/json"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+)
+
+var sampleConfig = `
+  ## It requires a url name.
+  ## Will be transmitted telegraf metrics to the HTTP Server using the below URL.
+  ## Note that not support the HTTPS.
+  url = "http://127.0.0.1:8080/metric"
+  ## http_headers option can add a custom header to the request.
+  ## The value is written as a delimiter(:).
+  ## Content-Type is required http header in http plugin.
+  ## so content-type of HTTP specification (plain/text, application/json, etc...) must be filled out.
+  http_headers = [ "Content-Type:application/json" ]
+  ## With this HTTP status code, the http plugin checks that the HTTP request is completed normally.
+  ## As a result, any status code that is not a specified status code is considered to be an error condition and processed.
+  expected_status_codes = [ 200, 204 ]
+  ## Configure response header timeout in seconds. Default : 3
+  response_header_timeout = 3
+  ## Configure dial timeout in seconds. Default : 3
+  dial_timeout = 3
+  ## Option to set the number of metrics to include in the request body.
+  buffer_limit = 10
+
+  ## Data format to output.
+  ## Each data format has it's own unique set of configuration options, read
+  ## more about them here:
+  ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_OUTPUT.md
+  data_format = "influx"
+`
+
+const (
+	POST = "POST"
+
+	DEFAULT_RESPONSE_HEADER_TIMEOUT = 3
+	DEFAULT_DIAL_TIME_OUT           = 3
+	DEFAULT_BUFFER_LIMIT            = 10
+)
+
+type Http struct {
+	// http required option
+	URL                 string   `toml:"url"`
+	HttpHeaders         []string `toml:"http_headers"`
+	ExpectedStatusCodes []int    `toml:"expected_status_codes"`
+
+	// Option with http default value
+	ResponseHeaderTimeout int `toml:"response_header_timeout"`
+	DialTimeOut           int `toml:"dial_timeout"`
+	BufferLimit           int `toml:"limit"`
+
+	client     http.Client
+	serializer serializers.Serializer
+	// expectedStatusCodeMap that stores option values received with expected_status_codes
+	expectedStatusCodeMap map[int]bool
+
+	// Context for request cancel of client
+	cancelContext context.Context
```

Comment:
I don't understand why you've used contexts, it doesn't seem like you're actually using them? and why have you made them a part of the Http struct?